### DISCUSSION TITLE
Could we make `isLongPressed()` public?

### DISCUSSION THREAD

**Discussion by Turbine1991** [bxparks/AceButton#94] [General]
Hi,
I'm overhauling my indicator system in my scooter. It has two buttons for indication.

I cycle through modes by tapping both indicators at the same time. However, I need to process combination taps but ignore it if either is long pressed. Rather than keep my own state tracker, why not make `isLongPressed()` public?

```
      if (eventType == AceButton::kEventPressed) {
        if (left.isPressedRaw() && right.isPressedRaw()) { // If both bottons are pressed
          if (!left.isLongPressed() && !right.isLongPressed()) { // Skip if either button is long pressed, taps only
            Serial.println("Both");
          }
        }
      }
```

**Comment by bxparks** [comment]
Hi, I think I need to understand your usage of the library better. Can you explain your button states in more detail? You got 2 buttons, left and right, and I gather you handle various Pressed and Long Pressed from these buttons individually. How exactly do you want the various combinations of both buttons being pressed to be handled?

To be frank, AceButton was designed to be a callback library (i.e. you register a callback function for specific events), instead of a polling library (i.e. you call functions on the buttons to check if certain conditions have happened). You seem to be using AceButton as a polling library, which it was not designed to be. I'm hesitant to expose the `isLongPressed()` or other methods that expose the internal states of the `AceButton` object because that would expand the complexity and surface area of the API. Without substantial amount of thinking and unit testing, I would not able to guarantee that those functions would be free of subtle bugs.
